tradeexecutor/cli/commands/perform_test_trade.py

```diff
@@ -106,30 +112,69 @@ def perform_test_trade(
 
     web3config.choose_single_chain()
 
-    execution_model, sync_model, valuation_model_factory, pricing_model_factory = create_execution_and_sync_model(
-        asset_management_mode=asset_management_mode,
-        private_key=private_key,
-        web3config=web3config,
-        confirmation_timeout=datetime.timedelta(seconds=60),
-        confirmation_block_count=confirmation_block_count,
-        min_gas_balance=min_gas_balance,
-        max_slippage=max_slippage,
-        vault_address=vault_address,
-        vault_adapter_address=vault_adapter_address,
-        vault_payment_forwarder_address=vault_payment_forwarder_address,
-        routing_hint=mod.trade_routing,
-    )
+    # assert len(mod.trade_routing) == 1, "Test trade only works with single routing strategies for now"
+
+    confirmation_timeout = datetime.timedelta(seconds=60)
+
+    generic_routing_data = None
+    if len(mod.trade_routing) > 1:
+        generic_routing_data, sync_model = create_generic_execution_and_sync_model(
```

Comment:
Please see the first comment

Reply:
There is no need to return multiple values.

You need only one execution model, sync model, valuation model and pricing model per strategy. I don't understand why there would be several, but maybe I am missing something here?

Reply:
It seems to me that: 

- the `sync model` can be shared between different dexes in a strategy (only need 1 per strategy)
- the `execution model` can only be shared between dexes of the same type. E.g. if using both Uniswap V2 and V3 exchanges, will need two execution models (`UniswapV2ExecutionModel` and `UniswapV3ExecutionModel`)
- Same for valuation model and pricing model as the execution model 
- Each dex will also need its own `routing_model`

Reply:
> the execution model can only be shared between dexes of the same type. E.g. if using both Uniswap V2 and V3 exchanges, will need two execution models (UniswapV2ExecutionModel and UniswapV3ExecutionModel)

Then the solution is to fix execution model so that you have GenericExecutionModel.